### PR TITLE
Fix Rating component

### LINKED ISSUE #54: Rating Component divides incoming rating by two
https://github.com/marwinburesch/moviescout/blob/fe44607d496f5c764a15b87bf4edf50a161570b1/src/app/components/Rating/Rating.tsx#L12

This shouldn't be done here in the component itself, but rather at least a level higher up

### PR DESCRIPTION
**Pull request by marwinburesch**
fixes #54 
The component should be a controlled component, it had too much control over the incoming prop `rating`

### CODE REVIEW

File: src/app/components/Rating/Rating.tsx [hunk 1]
```diff
@@ -9,27 +9,23 @@ function Rating({
...
-        {Array.from({ length: maxRating }, (_, i) => i + 1).map(
-          (value, index) => {
-            let starStatus: 'active' | 'inactive' | 'halfActive';
-            if (value < rating + 1) {
-              if (rating % 1 === 0 || rating > value) {
-                starStatus = 'active';
-              } else {
-                starStatus = 'halfActive';
-              }
+        {Array.from({ length: maxRating }, (_, i) => i + 1).map((value) => {
```
Comment (mpagels) [inline src/app/components/Rating/Rating.tsx:16]: Very nice. But hard to understand. Would it be a good idea to write a helper function that abstract this part?:
```suggestion
        {getRangeCountFrom(maxRating).map((value) => {
```

**Review by mpagels — APPROVED** [review]
The `Array.from()` logic is quite hard to understand quickly. Left an idea for that. Very nice fix.

File: src/app/components/Rating/Rating.tsx (hunk 1 above)
Comment (marwinburesch) [reply src/app/components/Rating/Rating.tsx:16]: Cheers dude, good idea!

**Review by mpagels — APPROVED** [review]
Noice!

**Review by phlgr — APPROVED** [review]

**Pull request merged**